### DISCUSSION TITLE
How far off is native ESM support (no Babel)

### DISCUSSION THREAD

**Discussion by shamilovtim** [facebook/hermes#1391] [Q&A] [answered]
Hi,

I have a question for my own curiosity. Is there a future where Hermes will no longer require Babel and will natively support ESM + latest ES?

**Comment by tmikov** [comment, marked as answer]
Yes, absolutely, and probably very soon. That is only way Static Hermes can reach its full performance potential. The main challenge which is holding back native ESM support is integrating this with the existing RN bundlers and build process.

**Reply by dannyhw** [reply]
sorry to randomly bump this but is there any update on esm support?

**Reply by tmikov** [reply]
No update for now. ESM is essential for cross-module optimizations in a AOT compiler, but at this time we unfortunately don't have a public roadmap for getting there. As I have mentioned before, one of the challenges is integrating with existing RN bundlers, which provide their own ESM emulation.
